File: sub-model/src/main/java/org/github/base/entity/SysUserOnlineEntity.java
package org.github.base.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import org.github.base.AbstractEntity;
import com.baomidou.mybatisplus.annotation.TableId;
import java.time.LocalDateTime;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.*;

/**
 * <p>
 * 在线用户记录
 * </p>
 *
 * @author JYD_XL
 * @since 2019-11-20
 */
@Setter
@Getter
@EqualsAndHashCode(callSuper = true)
@TableName("sys_user_online")
public class SysUserOnlineEntity extends AbstractEntity {
    /** UID */
    private static final long serialVersionUID = 1L;

    /** 用户会话id */
    @TableId("sessionId")
    private String sessionId;

    /** 登录账号 */
    @TableField("login_name")
    private String loginName;

    /** 部门名称 */
    @TableField("dept_name")
    private String deptName;

    /** 登录IP地址 */
    @TableField("ipaddr")
    private String ipaddr;

    /** 登录地点 */
    @TableField("login_location")
    private String loginLocation;

    /** 浏览器类型 */
    @TableField("browser")
    private String browser;

    /** 操作系统 */
    @TableField("os")
    private String os;

    /** 在线状态on_line在线off_line离线 */
    @TableField("status")
    private String status;

    /** session创建时间 */
    @TableField("start_timestamp")
    private LocalDateTime startTimestamp;

    /** session最后访问时间 */
    @TableField("last_access_time")
    private LocalDateTime lastAccessTime;

    /** 超时时间，单位为分钟 */
    @TableField("expire_time")
    private Integer expireTime;
}

File: sub-model/src/main/java/org/github/base/entity/SysUserOnlineEntity.kt
package org.github.base.entity

import com.baomidou.mybatisplus.annotation.TableName
import org.github.base.AbstractEntity
import com.baomidou.mybatisplus.annotation.TableId
import java.time.LocalDateTime

/**
 * <p>
 * 在线用户记录
 * </p>
 *
 * @author JYD_XL
 * @since 2019-11-19
 */
@TableName("sys_user_online")
class SysUserOnlineEntity: AbstractEntity() {

    /** 用户会话id */
    @TableId("sessionId")
    var sessionId: String? = null

    /** 登录账号 */
    var loginName: String? = null

    /** 部门名称 */
    var deptName: String? = null

    /** 登录IP地址 */
    var ipaddr: String? = null

    /** 登录地点 */
    var loginLocation: String? = null

    /** 浏览器类型 */
    var browser: String? = null

    /** 操作系统 */
    var os: String? = null

    /** 在线状态on_line在线off_line离线 */
    var status: String? = null

    /** session创建时间 */
    var startTimestamp: LocalDateTime? = null

    /** session最后访问时间 */
    var lastAccessTime: LocalDateTime? = null

    /** 超时时间，单位为分钟 */
    var expireTime: Int? = null

}
